tests/test_record.py

```diff
@@ -23,17 +24,46 @@ def start_requests(self):
 
     def parse(self, response):
         {parse}
+
+    def parse_item(self):
+        {parse_item}
 '''
 
 
 def run(*pargs, **kwargs):
-    proc = subprocess.Popen(*pargs, **kwargs)
-    proc.wait()
-    return {
-        'returncode': proc.returncode,
-        'stdout': proc.stdout.read(),
-        'stderr': proc.stderr.read(),
-    }
+    try:
+        process = subprocess.run(*pargs, **kwargs)
+        return {
+            'returncode': process.returncode,
+            'stdout': process.stdout,
+            'stderr': process.stderr,
+        }
+    except AttributeError:
+        input = kwargs.pop('input', None)
+        check = kwargs.pop('check', False)
+        # Python 2.7 compatibility
+        if input is not None:
+            if 'stdin' in kwargs:
+                raise ValueError(
+                    'stdin and input arguments may not both be used.')
+            kwargs['stdin'] = subprocess.PIPE
+
+        process = subprocess.Popen(*pargs, **kwargs)
+        try:
+            stdout, stderr = process.communicate(input)
+        except Exception:
+            process.kill()
+            process.wait()
+            raise
+        retcode = process.poll()
+        if check and retcode:
+            raise subprocess.CalledProcessError(
+                retcode, process.args, output=stdout, stderr=stderr)
+        return {
+            'returncode': retcode,
+            'stdout': stdout,
+            'stderr': stderr,
+        }
```

Comment:
The improvements to the tests here are awesome but it would be good to have them in a separate PR just not to have them mixed with changes that are more relevant to the issue that the PR tries to solve.

Reply:
ok! I added this part (l41-l66) to emulate subprocess.run for python2.7, if not travis would raise an error for python2.7.

Reply:
@mabelvj thanks for looking into the warnings issue.
They seem to be resources not closed and more specifically, looks like the issue is that we're calling `proc.stdout.read()` and never close that resource.
Closing `proc.stdout` afterwards or using a context manager around the `read()` calls should solve the problem. The same goes for `proc.stderr`.
Doing this, maybe we could still use Popen since works in both Py2 and Py3 and we'll be using less code.

Reply:
By the way @mabelvj, I think this might be a simpler approach to resolve the resource warnings issue https://github.com/scrapinghub/scrapy-autounit/pull/40.
Maybe we can skip this improvement from this PR and keep the focus on the spider attributes changes.

Reply:
Thanks for the suggestion @fcanobrash. Works perfectly! Just added it.